internal/provider/iam_inline_policy_resource.go

```diff
@@ -0,0 +1,406 @@
+				MarkdownDescription: "List of IAM inline policies to associate with the entity.",
+				Required:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Description:         "Name of the IAM inline policy.",
+							MarkdownDescription: "Name of the IAM inline policy.",
+							Required:            true,
+						},
+						"document": schema.StringAttribute{
+							Description:         "Policy document in JSON format.",
+							MarkdownDescription: "Policy document in JSON format.",
+							Required:            true,
+							CustomType:          jsontypes.NormalizedType{},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// ValidateConfig validates the resource configuration.
+func (r *IAMInlinePolicyResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var config models.IAMInlinePolicyResourceModel
+
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	count := 0
+	if !config.Username.IsNull() && !config.Username.IsUnknown() {
+		count++
+	}
+	if !config.Groupname.IsNull() && !config.Groupname.IsUnknown() {
+		count++
+	}
+	if !config.Rolename.IsNull() && !config.Rolename.IsUnknown() {
+		count++
+	}
+
+	if count == 0 {
+		resp.Diagnostics.AddError(
+			"Validation Error",
+			"Exactly one of username, groupname, or rolename must be provided.",
+		)
+	} else if count > 1 {
+		resp.Diagnostics.AddError(
+			"Validation Error",
+			"Only one of username, groupname, or rolename can be provided.",
+		)
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *IAMInlinePolicyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state models.IAMInlinePolicyResourceModel
+
+	// Get current state
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Determine namespace
+	namespace := state.Namespace.ValueString()
+
+	// Determine entity type and name
+	var entityType, entityName string
+	if !state.Username.IsNull() && !state.Username.IsUnknown() {
+		entityType = "User"
+		entityName = state.Username.ValueString()
+	} else if !state.Groupname.IsNull() && !state.Groupname.IsUnknown() {
+		entityType = "Group"
+		entityName = state.Groupname.ValueString()
+	} else if !state.Rolename.IsNull() && !state.Rolename.IsUnknown() {
+		entityType = "Role"
+		entityName = state.Rolename.ValueString()
+	}
+
+	// Step 1: Call List<entity>Policies API
+	var policyNames []string
+	switch entityType {
+	case "User":
+		listResp, _, err := r.client.GenClient.IamApi.IamServiceListUserPolicies(ctx).
+			XEmcNamespace(namespace).
+			UserName(entityName).
+			Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Read Error", fmt.Sprintf("Failed to list policies: %s", err))
```

Comment:
```suggestion
			resp.Diagnostics.AddError("Read Error", fmt.Sprintf("Failed to list policies: %s", err.Error()))
```
Applicable to all instances. Otherwise, the error message looks bad.